lib/rules/sort-keys.js

```diff
@@ -0,0 +1,239 @@
+            default: true
+          }
+        },
+        additionalProperties: false
+      }
+    ]
+  },
+
+  create (context) {
+    // Parse options.
+    const options = context.options[1]
+    const order = context.options[0] || 'asc'
+
+    const ignoreGrandchildrenOf = (options && options.ignoreGrandchildrenOf) || ['computed', 'directives', 'inject', 'props', 'watch']
+    const ignoreChildrenOf = (options && options.ignoreChildrenOf) || ['model']
+    const insensitive = options && options.caseSensitive === false
+    const minKeys = options && options.minKeys
+    const natual = options && options.natural
+    const isValidOrder = isValidOrders[
+      order + (insensitive ? 'I' : '') + (natual ? 'N' : '')
+    ]
+
+    // The stack to save the previous property's name for each object literals.
+    let stack = null
+
+    let errors = []
+
+    const reportErrors = (isVue) => {
+      if (isVue) {
+        errors = errors.filter((error) => {
+          let filter = error.hasUpper
+          if (!error.grandparentName) {
+            filter = filter && !ignoreChildrenOf.includes(error.parentName)
+          } else if (!error.greatGrandparentName) {
+            filter = filter && !ignoreGrandchildrenOf.includes(error.grandparentName)
+          }
+          return filter
+        })
+      }
+      errors.forEach((error) => context.report(error))
+      errors = []
+    }
+
+    const sortTests = {
+      ObjectExpression (node) {
+        if (!stack) {
+          reportErrors(false)
+        }
+        stack = {
+          upper: stack,
+          prevName: null,
+          numKeys: node.properties.length
+        }
+      },
+      'ObjectExpression:exit' () {
+        stack = stack.upper
+      },
+      SpreadElement (node) {
+        if (node.parent.type === 'ObjectExpression') {
+          stack.prevName = null
+        }
+      },
+      'Program:exit' () {
+        reportErrors(false)
+      },
+      Property (node) {
+        if (node.parent.type === 'ObjectPattern') {
+          return
+        }
+
+        const prevName = stack.prevName
+        const numKeys = stack.numKeys
+        const thisName = getPropertyName(node)
+
+        if (thisName !== null) {
+          stack.prevName = thisName
+        }
+
+        if (prevName === null || thisName === null || numKeys < minKeys) {
+          return
+        }
+
+        if (!isValidOrder(prevName, thisName)) {
+          errors.push({
+            node,
+            hasUpper: !!stack.upper,
+            loc: node.key.loc,
+            message: "Expected object keys to be in {{natual}}{{insensitive}}{{order}}ending order. '{{thisName}}' should be before '{{prevName}}'.",
+            parentName: stack.upper && stack.upper.prevName,
+            grandparentName: stack.upper && stack.upper.upper && stack.upper.upper.prevName,
+            greatGrandparentName: stack.upper && stack.upper.upper && stack.upper.upper.upper && stack.upper.upper.upper.prevName,
```

Comment:
It may false negatives if the objects are not chained.

```js
export default {
  computed: {
    foo () {
      return {
        b,
        a
      }
    }
  }
}
```